src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1927 @@
+                        case 'n': ch = '\n'; break;
+                        case 'r': ch = '\r'; break;
+                        case 't': ch = '\t'; break;
+                        case 'v': ch = '\u000B'; break;
+                    }
+                    return true;
+
+                case RegexKind.ControlEscape:
+                    var controlEscape = (RegexControlEscapeNode)component;
+                    var controlCh = controlEscape.ControlToken.VirtualChars[0].Char;
+                    // \ca interpreted as \cA
+                    if (controlCh >= 'a' && controlCh <= 'z')
+                    {
+                        controlCh -= (char)('a' - 'A');
+                    }
+                    ch = (char)(controlCh - '@');
+                    return true;
+
+                case RegexKind.OctalEscape:
+                    ch = GetCharValue(((RegexOctalEscapeNode)component).OctalText, withBase: 8);
+                    return true;
+
+                case RegexKind.HexEscape:
+                    ch = GetCharValue(((RegexHexEscapeNode)component).HexText, withBase: 16);
+                    return true;
+
+                case RegexKind.UnicodeEscape:
+                    ch = GetCharValue(((RegexUnicodeEscapeNode)component).HexText, withBase: 16);
+                    return true;
+
+                case RegexKind.PosixProperty:
+                    // When the native parser sees [:...:] it treats this as if it just saw '[' and skipped the 
+                    // rest.
+                    ch = '[';
+                    return true;
+
+                case RegexKind.Text:
+                    ch = ((RegexTextNode)component).TextToken.VirtualChars[0];
+                    return true;
+
+                case RegexKind.Sequence:
+                    var sequence = (RegexSequenceNode)component;
+#if DEBUG
+                    Debug.Assert(sequence.ChildCount > 0);
+                    for (int i = 0, n = sequence.ChildCount - 1; i < n; i++)
+                    {
+                        Debug.Assert(IsEscapedMinus(sequence.ChildAt(i).Node));
+                    }
+#endif
+
+                    var last = sequence.ChildAt(sequence.ChildCount - 1).Node;
+                    if (IsEscapedMinus(last))
+                    {
+                        break;
+                    }
+
+                    return TryGetRangeComponentValueWorker(last, out ch);
+            }
+
+            ch = default;
+            return false;
+        }
+
+        private char GetCharValue(RegexToken hexText, int withBase)
+        {
+            unchecked
+            {
+                var total = 0;
+                foreach (var vc in hexText.VirtualChars)
+                {
+                    total *= withBase;
+                    total += HexValue(vc.Char);
+                }
+
+                return (char)total;
+            }
+        }
+
+        private int HexValue(char ch)
+        {
+            unchecked
```

Comment:
I'm having trouble imagining why this `unchecked` matters. Characters others than `0-9`, `a-z` and `A-Z` can't make it into a hex token, right?

Reply:
That's a good point.  I think i was just being paranoid.  Since i was doing mathematics, i dind't want to accidently trigger anything.  Given that the max unicode/hexvalue we'll read in is \uFFFF, and this is only 16 bits, being unchcked shouldn't matter.  will remove and will add test.